src/3d/mesh/qgsmesh3dentity_p.cpp

```diff
@@ -0,0 +1,125 @@
+/***************************************************************************
+                         qgsmesh3dentity.cpp
+                         -------------------------
+    begin                : january 2020
+    copyright            : (C) 2020 by Vincent Cloarec
+    email                : vcloarec at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsmesh3dentity_p.h"
+
+#include <QOpenGLContext>
+#include <qopenglfunctions_3_2_core.h>
+
+#include <Qt3DRender/QTexture>
+#include <Qt3DRender/QParameter>
+#include <Qt3DExtras/QTextureMaterial>
+#include <Qt3DExtras/QPhongMaterial>
+#include <Qt3DExtras/QMetalRoughMaterial>
+
+#include "qgsmeshlayer.h"
+#include "qgsmapsettings.h"
+#include "qgs3dmapsettings.h"
+#include "qgsmeshlayerrenderer.h"
+#include "qgsterraintextureimage_p.h"
+#include "qgsmesh3dmaterial_p.h"
+
+
+QgsMesh3dEntity::QgsMesh3dEntity( const Qgs3DMapSettings &map,
+                                  const QgsTriangularMesh triangularMesh,
+                                  const QgsRectangle &extent,
+                                  const QgsMesh3DSymbol &symbol ):
+  mExtent( extent ),
+  mSymbol( symbol ),
+  mMapSettings( map ),
+  mTriangularMesh( triangularMesh )
+{}
+
+void QgsMesh3dEntity::build()
+{
+  buildGeometry();
+  applyMaterial();
+}
+
+void QgsMesh3dEntity::buildGeometry()
+{
+  Qt3DRender::QGeometryRenderer *mesh = new Qt3DRender::QGeometryRenderer;
+  QVector<double> fakeScalarMag( mTriangularMesh.vertices().count() );
+  for ( int i = 0; i < mTriangularMesh.vertices().count(); ++i )
+  {
+    fakeScalarMag[i] = mTriangularMesh.vertices().at( i ).z();
+  }
+
+#if 0
+  mesh->setGeometry( new QgsMesh3dGeometry_p( mTriangularMesh,
+                     mExtent,
+                     mSymbol.verticaleScale(),
+                     mesh ) );
+#else
```

Comment:
Can we remove this part - and maybe remove the whole geometry implementation as well? From a quick look the difference between the two geometry implementations is small, so better to keep just one and improve it later (so that e.g. the "magnitude" attribute would be optional if such optimization would be worth the effort